### DISCUSSION TITLE
[Proposal] Simplifying inheritance of base class constructors

### DISCUSSION THREAD

**Discussion by amitchneck** [dotnet/csharplang#5732] [Language Ideas]
In C#, to remove boilerplate code, it would be useful to add `..` operator to ctor operations.

The `..` operator, when used in base call, would call the base constructor whose signature matches the calling child class's constructor.
```
class A
{
  public A() { }
  public A(string s) { }
  public A(int i, string s, char c) { }
}
class B : A
{
  public B() : base(..) // same as base()
  public B(string s) : base(..) // same as base(s)
  public B(int i, string s, char c) : base(..) // same as base(i, s, c)
}
```
The `..` operator, when used in constructor, automatically inherits all base class constructors, unless constructor with similar signature already exists. Constructor using `..` operator is considered to have no input parameters for statements/expressions written within its context.
```
class C : A
{
  int j;
  public C(..)
  {
    // there are no parameters available for code written here
    j = 5;
  }
  /*
  the above is the same as writing:
  public C() : base()
  {
    j = 5;
  }
  public C(string s) : base(s)
  {
    j = 5;
  }
  public C(int i, string s, char c) : base(i, s, c)
  {
    j = 5;
  }
  */
}

class D : A
{
  int j;
  public D(string s) : base(..)
  {
    j = 6;
  }
  public D(..)
  {
    // there are no parameters available for code written here
    j = 3;
  }
  /*
  the above is the same as writing:
  public D() : base()
  {
    j = 3;
  }
  public D(int i, string s, char c) : base(i, s, c)
  {
    j = 3;
  }
  notice public D(string s) is not included as it is defined elsewhere in this class
  */
}
```
Not only does this simplify code, it removes unnecessary constructors created just to inherit all base class constructors.

Any thoughts on the matter?

**Comment by TahirAhmadov** [comment]
Check out my proposal here: https://github.com/dotnet/csharplang/discussions/5601

**Comment by CyrusNajmabadi** [comment]
`base (..)` is already legal c# code that has meaning today.

**Reply by amitchneck** [reply]
I have never seen this and can't find it on Google. What is the current meaning of `base(..)`?

**Reply by HaloFour** [reply]
`..` is an unbounded range.

**Reply by amitchneck** [reply]
I see .... The specific operator can be whatever; I'm more looking at the concept. Since `..` is used for indexing/ranging, how about making the operator an underscore `_`?

**Reply by MgSam** [reply]
Mads had proposed this at one point inside of a much larger proposal (I forget which one). Sadly I don't think it was ever broken out into its own feature.
